src/Client.php

```diff
@@ -148,15 +147,15 @@ public function setRawDataStream($data, $enctype = null)
      */
     public function setMethod($method = HttpRequest::METHOD_GET)
     {
-        if ($method === HttpRequest::METHOD_GET) {
+        if (strtolower($method) === strtolower(HttpRequest::METHOD_GET)) {
             $this->setRequestMethod(HttpRequest::METHOD_GET);
-        } elseif ($method === HttpRequest::METHOD_POST) {
+        } elseif (strtolower($method) === strtolower(HttpRequest::METHOD_POST)) {
             $this->setRequestMethod(HttpRequest::METHOD_POST);
-        } elseif ($method === HttpRequest::METHOD_PUT) {
+        } elseif (strtolower($method) === strtolower(HttpRequest::METHOD_PUT)) {
             $this->setRequestMethod(HttpRequest::METHOD_PUT);
-        } elseif ($method === HttpRequest::METHOD_DELETE) {
+        } elseif (strtolower($method) === strtolower(HttpRequest::METHOD_DELETE)) {
             $this->setRequestMethod(HttpRequest::METHOD_DELETE);
-        } elseif ($method === HttpRequest::METHOD_HEAD) {
+        } elseif (strtolower($method) === strtolower(HttpRequest::METHOD_HEAD)) {
```

Comment:
The constants in `HttpRequest` are all uppercase - just one call such as `$method = strtoupper($method);` is sufficient rather than transforming everything twice for each branch, or even changing it to `switch(strtoupper($method))`